collector.go

```diff
@@ -98,38 +154,73 @@ func (c Collector) measure() error {
 		for time.Since(begin) < c.NtpMeasurementDuration {
-			nextClockOffset, nextStratum, err := c.getClockOffsetAndStratum()
+			nextClockOffset, nextStratum, nextRTT, nextReferenceTime, nextRootDelay, nextRootDispersion, nextRootDistance, nextPrecision, nextLeap, err := c.getClockOffsetAndStratum()
 			if err != nil {
 				return fmt.Errorf("couldn't get NTP drift: %s", err)
 			}
 
 			measurementsClockOffset = append(measurementsClockOffset, nextClockOffset)
 			measurementsStratum = append(measurementsStratum, nextStratum)
+			measurementsRTT = append(measurementsRTT, nextRTT)
+			measurementsReferenceTime = append(measurementsReferenceTime, nextReferenceTime)
+			measurementsRootDelay = append(measurementsRootDelay, nextRootDelay)
+			measurementsRootDispersion = append(measurementsRootDispersion, nextRootDispersion)
+			measurementsRootDistance = append(measurementsRootDispersion, nextRootDistance)
+			measurementsPrecision = append(measurementsPrecision, nextPrecision)
+			measurementsLeap = append(measurementsLeap, nextLeap)
 		}
 
 		clockOffset = calculateMedian(measurementsClockOffset)
 		stratum = calculateMedian(measurementsStratum)
+		rtt = calculateMedian(measurementsRTT)
+		referenceTime = calculateMedian(measurementsReferenceTime)
+		rootDelay = calculateMedian(measurementsRootDelay)
+		rootDispersion = calculateMedian(measurementsRootDispersion)
+		rootDistance = calculateMedian(measurementsRootDistance)
+		precision = calculateMedian(measurementsPrecision)
+		leap = calculateMedian(measurementsLeap)
 	}
 
 	c.drift.WithLabelValues(c.NtpServer).Set(clockOffset)
-	c.stratum.Set(stratum)
+	c.stratum.WithLabelValues(c.NtpServer).Set(stratum)
+	c.rtt.WithLabelValues(c.NtpServer).Set(rtt)
+	c.referenceTime.WithLabelValues(c.NtpServer).Set(referenceTime)
+	c.rootDelay.WithLabelValues(c.NtpServer).Set(rootDelay)
+	c.rootDispersion.WithLabelValues(c.NtpServer).Set(rootDispersion)
+	c.rootDistance.WithLabelValues(c.NtpServer).Set(rootDistance)
+	c.precision.WithLabelValues(c.NtpServer).Set(precision)
+	c.leap.WithLabelValues(c.NtpServer).Set(leap)
 
 	c.scrapeDuration.Observe(time.Since(begin).Seconds())
 	return nil
 }
 
-func (c Collector) getClockOffsetAndStratum() (clockOffset float64, strat float64, err error) {
+func (c Collector) getClockOffsetAndStratum() (clockOffset float64, strat float64, rtt float64, referenceTime float64, rootDelay float64, rootDispersion float64, rootDistance float64, precision float64, leap float64, err error) {
 	options := ntp.QueryOptions{Version: c.NtpProtocolVersion}
 	resp, err := ntp.QueryWithOptions(c.NtpServer, options)
 	if err != nil {
-		return 0, 0, fmt.Errorf("couldn't get NTP drift: %s", err)
+		return 0, 0, 0, 0, 0, 0, 0, 0, 0, fmt.Errorf("couldn't get NTP drift: %s", err)
 	}
 	clockOffset = resp.ClockOffset.Seconds()
+	rtt = resp.RTT.Seconds()
+	referenceTime = float64(time.Duration(resp.ReferenceTime.Unix()))
```

Comment:
Why do you go through `time.Duration` here? Should it not be enough to go from int64 to float64 directly?

Reply:
Thanks, fixed!